packages/js/src/constants.ts

```diff
@@ -0,0 +1,31 @@
+/**
+ * Code Verifier - Characters Range
+ *
+ * The code verifier is a string using the unreserved characters [A-Z] / [a-z] / [0-9] / "-" / "." / "_" / "~" ( url-safe ) .
+ * @link [Client Creates a Code Verifier](https://datatracker.ietf.org/doc/html/rfc7636#section-4.1)
+ */
+export const CODE_VERIFIER_ALPHABET =
+  'ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789-._~';
+
+/**
+ * Code Verifier - Length Range
+ *
+ * The length of code verifier ranges from 43 to 128.
+ * @link [Client Creates a Code Verifier](https://datatracker.ietf.org/doc/html/rfc7636#section-4.1)
+ */
+export const CODE_VERIFIER_MIN_LENGTH = 43;
+export const CODE_VERIFIER_MAX_LENGTH = 128;
+
+/**
+ * Scope - Values
+ */
+export const NAME = 'name';
+export const EMAIL = 'email';
+export const OFFLINE_ACCESS = 'offline_access';
+export const OPENID = 'openid';
+
+/**
+ * Scope - Defaults
+ */
+export const DEFAULT_SCOPE_STRING = `${OPENID} ${OFFLINE_ACCESS}`;
```

Comment:
we should be able to generate the stringified scope by referencing  `DEFAULT_SCOPE_VALUES` or the other way around depends on how we gonna consume the default scopes.   But should keep a single source of truth, should not keep two different types of references of default scopes.

Reply:
prefer: generate the stringified scope by referencing DEFAULT_SCOPE_VALUES